apps/asap-server/src/entities/team.ts

```diff
@@ -43,7 +43,15 @@ export const parseGraphQLTeam = (
   const outputs: ResearchOutputResponse[] = flatOutputs
     .map((o) => {
       const output = parseGraphQLResearchOutput(o as GraphqlResearchOutput);
-      return { ...output, team: { id: team.id, displayName } };
+
+      return {
+        ...output,
+        team: { id: team.id, displayName },
+        teams: (o as GraphqlResearchOutput).flatData?.teams?.map((t) => ({
```

Comment:
this is where we either grab the `teams` response when they are present or otherwise use the parent `team`. 

For Group response for instance no `teams` are being fetched so only a single team will always be returned as part of the `teams` property (even though more might be present in the DB)